roms_tools/plot.py

```diff
@@ -597,13 +631,464 @@ def _add_field_to_ax(
+        lat_deg = grid_ds["lat_coarse"].rename(
+            {"eta_coarse": "eta_rho", "xi_coarse": "xi_rho"}
+        )
+        lon_deg = grid_ds["lon_coarse"].rename(
+            {"eta_coarse": "eta_rho", "xi_coarse": "xi_rho"}
+        )
+        mask = grid_ds["mask_coarse"].rename(
+            {"eta_coarse": "eta_rho", "xi_coarse": "xi_rho"}
+        )
+    else:
+        lat_deg = grid_ds[f"lat_{loc}"]
+        lon_deg = grid_ds[f"lon_{loc}"]
+        mask = grid_ds[f"mask_{loc}"]
+
+    if straddle:
+        lon_deg = xr.where(lon_deg > 180, lon_deg - 360, lon_deg)
+    if lon is not None:
+        lon = normalize_longitude(lon, straddle)
+
+    field = field.assign_coords({"lon": lon_deg, "lat": lat_deg})
+
+    # Mask the field
+    field = field.where(mask)
+
+    # Assign eta and xi as coordinates
+    coords_to_assign = {dim: field[dim] for dim in horizontal_dims.values()}
+    field = field.assign_coords(**coords_to_assign)
+
+    # Remove horizontal boundary if desired
+    slice_dict = {
+        "rho": {"eta_rho": slice(1, -1), "xi_rho": slice(1, -1)},
+        "u": {"eta_rho": slice(1, -1), "xi_u": slice(1, -1)},
+        "v": {"eta_v": slice(1, -1), "xi_rho": slice(1, -1)},
+    }
+    if not include_boundary:
+        field = field.isel(**slice_dict[loc])
+
+    # Prepare figure title
+    if "abs_time" in field.coords:
+        formatted_time = np.datetime_as_string(field.abs_time.values, unit="m")
+        title = f"time: {formatted_time}"
+    elif "ntides" in field.coords:
+        title = f"constituent: {field.ntides.values.item().decode('utf-8')}"
+    elif "long_name" in field.attrs:
+        title = field.long_name
+    else:
+        title = ""
+
+    # Slice the field horizontally as desired
+    def _slice_along_dimension(field, title, dim_name, idx):
+        field = field.sel(**{dim_name: idx})
+        title = title + f", {dim_name} = {idx}"
+        return field, title
+
+    if eta is not None:
+        field, title = _slice_along_dimension(field, title, horizontal_dims["eta"], eta)
+    if xi is not None:
+        field, title = _slice_along_dimension(field, title, horizontal_dims["xi"], xi)
+    if s is not None:
+        field, title = _slice_along_dimension(field, title, "s_rho", s)
+
+    # Compute layer and interface depth if necessary
+    compute_layer_depth = ("s_rho" in field.dims) or (
+        depth_contours and s is not None and lat is None and lon is None
+    )
+    compute_interface_depth = (
+        len(field.dims) == 2
+        and "s_rho" in field.dims
+        and layer_contours
+        and (lat is None)
+        and (lon is None)
+        and (depth is None)
+    )
+
+    def _compute_and_slice_depth(depth_type):
+        depth = compute_depth_coordinates(
+            grid_ds, zeta, depth_type=depth_type, location=loc, eta=eta, xi=xi
+        )
+        if not include_boundary:
+            depth = depth.isel(
+                **{
+                    dim: s
```

Comment:
Does it matter here that the `s is not None` check hasn't been tested like line `987`?

Reply:
Sorry, this is confusing. `s` is a general index here, not the `s` provided by the user. I'm changing this for clarity.